### PR TITLE
samples: peripheral: lpuart: use DEVICE_DT_GET

### PR DESCRIPTION
**Pull request by gmarull**
Obtain LPUART device at compile time using DEVICE_DT_GET. Also remove
init sleep of 1s, if this is papering over a bug it should be fixed
properly.

### CODE REVIEW

File: samples/peripheral/lpuart/src/main.c [hunk 1]
```diff
@@ -5,6 +5,8 @@
  */
 
 #include <zephyr.h>
+#include <device.h>
```
Comment (anangl) [inline samples/peripheral/lpuart/src/main.c:8]: This will anyway get included through `<drivers/uart.h>`.

**Review by anangl — APPROVED** [review]

**Review by mbolivar-nordic — CHANGES_REQUESTED** [review]
>  Also remove init sleep of 1s, if this is papering over a bug it should be fixed properly.

This should be a separate patch so it can be reverted standalone if necessary

File: samples/peripheral/lpuart/src/main.c (hunk 1 above)
Comment (gmarull) [reply samples/peripheral/lpuart/src/main.c:8]: I prefer to be explicit about header inclusion (we're using device.h APIs)

Comment (anangl) [reply samples/peripheral/lpuart/src/main.c:8]: I see. It just seems a bit redundant to me, as when a device driver is used, you by definition have the device API defined.

**Review by mbolivar-nordic — APPROVED** [review]

**Pull request merged**